lib/teleterm/services/desktop/desktop.go

```diff
@@ -28,23 +31,81 @@ import (
 	api "github.com/gravitational/teleport/gen/proto/go/teleport/lib/teleterm/v1"
 	"github.com/gravitational/teleport/lib/client"
 	"github.com/gravitational/teleport/lib/srv/desktop/tdp"
+	"github.com/gravitational/teleport/lib/teleterm/api/uri"
 )
 
-// Connect starts a remote desktop session.
-func Connect(ctx context.Context, stream grpc.BidiStreamingServer[api.ConnectToDesktopRequest, api.ConnectToDesktopResponse], clusterClient *client.TeleportClient, proxyClient *proxy.Client, desktopName, login string) error {
+// Session uniquely describes a desktop session.
+// There can be only one session for the given desktop and login pair.
```

Comment:
Are you relying on Windows to enforce this, or does Connect also make sure we only ever try to have a single session per (desktop, login) pair?

Reply:
Connect enforces this on two levels:
1. In the UI: if there is an active session for the given pair, attempting to open it again will switch you to the active session.
2. In tshd: attempting to open a new desktop session for the existing pair will return an error.